src/zope/interface/_zope_interface_coptimizations.c

```diff
@@ -325,6 +330,9 @@ Spec_clear(Spec* self)
 static void
 Spec_dealloc(Spec* self)
 {
+    /* PyType_GenericAlloc that you get when you don't
+       specify a tp_alloc always tracks the object. */
+    PyObject_GC_UnTrack((PyObject *)self);
```

Comment:
k, now I think this is option B: changes in this PR (specifically, specifying a custom tp_alloc instead of 0)?

Reply:
When I added C fields to `InterfaceBase`, it was necessary to untrack it. As it now extends `SpecificationBase` I was going up the method hierarchy and realized it was necessary also to untrack `SpecificationBase`, since it also has fields that can contain arbitrary objects. So it's a bit of both? But really a bug in https://github.com/zopefoundation/zope.interface/pull/155